subsys/net/ip/net_pkt.c

```diff
@@ -564,6 +567,12 @@ static struct net_pkt *net_pkt_get(struct k_mem_slab *slab,
 		}
 
 		pkt->data_len = data_len;
+	} else {
+		/*
+		 * In case, net_pkt_append() is later used on this packet,
+		 * set the RX packet data_len to a nice high number
+		 */
+		pkt->data_len = 0x1000;
```

Comment:
This we can't set, as we don't know the lenght of the incoming packet. pkt->data_len needs not be consulted on receiving, only when we attempt to send.

Reply:
The problem is that the DTLS receive path for net_app uses net_pkt_append() and it requires data_len be set if the context is set: https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/ip/net_pkt.c#L1220

Otherwise you get max_len 0.

Reply:
FYI: Here's the DTLS receive path in net_app where net_pkt_append is used: https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/app/net_app.c#L2087

Summary:
- Allocate the packet via net_pkt_get_rx() which calls net_pkt_get()
- Add IP UDP/TCP headers if found
- Append the decrypted data from request_buf via net_pkt_append_all() which calls net_pkt_append()

We would need the pkt->data_len to be set somewhere before the data append for it to work correctly.  Otherwise we can't set the net_context of the packet.